### DISCUSSION TITLE
Faulty SNZB-06P refuses to report

### DISCUSSION THREAD

**Discussion by mouldybread** [Koenkk/zigbee2mqtt#22874] [Devices]
Hi,

I have what I believe to be a faulty SNZB-06P. I've got 7 others working fine so I'm pretty sure it's the device. 

Here's the state from a working example:

`{
    "illumination": "bright",
    "linkquality": 156,
    "occupancy": false,
    "occupancy_sensitivity": "low",
    "occupancy_timeout": 15,
    "update": {
        "installed_version": 4102,
        "latest_version": 4101,
        "state": "idle"
    }
}`

And here's the state from the faulty device:

`
{
    "linkquality": 240,
    "occupancy_sensitivity": "low",
    "update": {
        "installed_version": 4102,
        "latest_version": 4101,
        "state": "idle"
    }
}`

Z2M can check the version and change settings fine but it never reports occupancy, so im guessing the daughter board with the radar is duff.

Before I bin it, is there a command I can enter to try and force it to report and confirm it's dead? Or anything I can try to get it working? TIA

**Comment by LeeStevens318** [comment]
I'm also having this issue on Firmware 1.0.6

```
{
    "linkquality": 255,
    "occupancy": true,
    "update": {
        "installed_version": 4102,
        "latest_version": 4101,
        "state": "idle"
    },
    "update_available": false
}
```

**Comment by mouldybread** [comment]
I gave up on mine. I did reflow the solder on the pins connecting the daughter board just in case but to no avail. The replacement worked fine.

My best guess, and I don't know enough to be sure, is that these devices are not well equipped to handle voltage fluctuations. They're not far off being battery powered and I don't see a great many components for regulating the supply. I am using Meanwell 5V packs to drive them which are probably kinder than most but every so often one or two get stuck, and this seems to coincide with a dip in infeed voltage. Perhaps it doesn't take much to fry the radar boards completely.

**Reply by nanderson97651** [reply]
Did you reach out to Sonoff support for replacements?  I just migrated from a SkyConnect to the SLZB-06 coordinator and neither of my devices report any updates to occupancy anymore.

```
{
    "illumination": null,
    "linkquality": 14,
    "occupancy": false,
    "occupancy_sensitivity": "medium",
    "occupancy_timeout": 60,
    "update": {
        "installed_version": 4102,
        "latest_version": 4102,
        "state": "idle"
    }
}
```

**Reply by mouldybread** [reply]
No, I just ordered my own to save the hassle. The unit I posted about was the only one that was completely useless. Sometimes when one stopped reporting I found that setting properties like occupancy_timeout would cause reporting to resume. I also found mine stopped reporting when I disconnected my repeater and they were being used as routers. I suspect the zigbee controller on these devices is just inherently weak and the inability to adjust the noise floor really limits their usefulness beyond very straight forward scenarios. 

I eventually switched to HLK LD2410Cs attached to an ESP8266 running ESPHOME. I already had 5v available and the api over wifi is lower latency. The app and bluetooth connectivity is great for in-situ tuning and they're rock solid, hidden in my led ceiling lights.

**Reply by nanderson97651** [reply]
Oh wow I'll look into those LD2410Cs.  Seems like Aliexpress has them for pretty cheap and I already run plenty of ESPHome devices.  Was there a yaml you used as reference?

In the meantime I'll poke around with the occupancy_timeout and try to re-pair them.  This all started when migrating to the new coordinator and rebuilding my network, so it may be related to the repeater vs router.  I even tried repairing one to the old skyconnect and it was exhibiting the same behavior.  So seems more likely a poor zigbee implementation rather than a specific issue with either either the Ember or Zstack stacks.

**Reply by mouldybread** [reply]
The LD2410 is [officially supported](https://esphome.io/components/sensor/ld2410.html) so I just went from there. I experimented with most of their mmWave range but things like lack of bluetooth, lack of support and/or instability ruled out other models for me. A company called Screek claim that a version of the LD2410 is coming with automatic noise floor adjustment but I couldn't find that version at the time.

The modules usually have a high/low output pin so strictly speaking you don't need support, you can just use a binary_sensor to monitor the pin and configure with bluetooth or uart.

The module inside the SLZB-06 looks a lot like a HLK module, maybe the HLK-LD012-5G. It does also seem to have an output pin so it should be possible to reuse them but the HLK modules are so cheap I opted not to bother.

Good luck!
